File: twister-lib-android/espresso_glide3/src/main/java/net/twisterrob/android/test/espresso/idle/GlideResetRule.java
package net.twisterrob.android.test.espresso.idle;

import org.junit.rules.ExternalResource;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.test.core.app.ApplicationProvider;

public class GlideResetRule extends ExternalResource {
	private final @NonNull Context appContext;

	public GlideResetRule() {
		this(ApplicationProvider.getApplicationContext());
	}

	public GlideResetRule(@NonNull Context appContext) {
		this.appContext = appContext;
	}

	@Override protected void before() {
		GlideResetter.resetGlide(appContext);
	}

	@Override protected void after() {
		GlideResetter.resetGlide(appContext);
	}
}

File: twister-lib-android/espresso_glide4/src/main/java/net/twisterrob/android/test/espresso/idle/GlideResetRule.kt
package net.twisterrob.android.test.espresso.idle

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import org.junit.rules.ExternalResource

open class GlideResetRule @JvmOverloads constructor(
	private val appContext: Context = ApplicationProvider.getApplicationContext()
) : ExternalResource() {

	override fun before() {
		GlideResetter.resetGlide(appContext)
	}

	override fun after() {
		GlideResetter.resetGlide(appContext)
	}
}
